apps/src/lab2/levelEditors/aiCustomizations/EditAiCustomizations.tsx

```diff
@@ -0,0 +1,112 @@
+import {
+  AiCustomizations,
+  LevelAiCustomizations,
+  Visibility,
+} from '@cdo/apps/aichat/types';
+import React, {useCallback, useState} from 'react';
+import {AppName} from '../../types';
+import {
+  BodyOneText,
+  BodyThreeText,
+} from '@cdo/apps/componentLibrary/typography';
+import SimpleDropdown from '@cdo/apps/componentLibrary/simpleDropdown/SimpleDropdown';
+import moduleStyles from './edit-ai-customizations.module.scss';
+
+interface EditAiCustomizationsProps {
+  initialCustomizations: LevelAiCustomizations;
+  levelName: string;
+  appName: AppName;
+}
+
+const EditAiCustomizations: React.FunctionComponent<
+  EditAiCustomizationsProps
+> = ({initialCustomizations, levelName, appName}) => {
+  const [aiCustomizations, setAiCustomizations] =
+    useState<LevelAiCustomizations>(initialCustomizations);
+
+  const setPropertyVisiblity = useCallback(
+    (property: keyof AiCustomizations, visibility: Visibility) => {
+      setAiCustomizations({
+        ...aiCustomizations,
+        [property]: {
+          value: aiCustomizations[property]?.value,
+          visibility,
+        },
+      });
+    },
+    [setAiCustomizations, aiCustomizations]
+  );
+
+  return (
+    <div>
+      <input
+        type="hidden"
+        id="level_initial_ai_customizations"
+        name="level[initial_ai_customizations]"
+        value={JSON.stringify(aiCustomizations)}
```

Comment:
For my understanding, is this a standard levelbuilder pattern (hidden input)?

Reply:
I think when I was working on the Lab2 validations editor, I saw this in a few other places and just copied it. I'm not sure if it's the ideal method, but we basically need a way for the levelbuilder edit page to grab the data in this component when we click the save button (i.e. submit the form). So this allows us to effectively split up the different pieces of the data into their own inputs, and then combine them back together into a single JSON object when we actually want to save it